www/include/eventLogs/viewLog.php

```diff
@@ -68,32 +68,42 @@
 $tpl = new Smarty();
 $tpl = initSmartyTpl("./include/eventLogs/template", $tpl);
 
-$engine = 'false';
-if (isset($_GET["engine"]) && $_GET["engine"] == 'true') {
-    $engine = 'true';
-}
-
-$output = "";
-if (isset($_GET["output"])) {
-    $output = $_GET["output"];
-}
-
-$openid = '0';
-if (isset($_GET["openid"])) {
-    $openid = $_GET["openid"];
-}
-
-$id = filter_var(
-    $_GET['id'] ?? $_POST['id'] ?? 1,
-    FILTER_VALIDATE_INT
+$filterParameters = [
+    'engine' => [
+        'filter' => FILTER_VALIDATE_BOOLEAN,
+        'options' => [
+            'default' => false
+        ]
+    ],
+    'id' => [
+        'filter' => FILTER_VALIDATE_INT,
+        'options' => [
+            'default' => 1
+        ],
+    ],
+    'h' => FILTER_SANITIZE_STRING,
+    'hg' => FILTER_SANITIZE_STRING,
+    'poller' => FILTER_SANITIZE_STRING,
+    'svc' => FILTER_SANITIZE_STRING,
+    'svcg' => FILTER_SANITIZE_STRING,
+    'output' => FILTER_SANITIZE_STRING,
+];
+
+$getInputs = filter_input_array(
+    INPUT_GET,
+    $filterParameters
+);
+$postInputs = filter_input_array(
+    INPUT_POST,
+    $filterParameters
 );
 
 $serviceGrpArray = array();
 $pollerArray = array();
 
 $defaultHosts = array();
-if (isset($_GET['h'])) {
-    $h = explode(",", $_GET['h']);
+if (isset($getInputs['h'])) {
+    $h = explode(",", $getInputs['h']);
```

Comment:
you also need to check that the array resulting from the explode contains only integers
eg : '12, "toto", 99, "\;sqlInjection"

Reply:
done